{{ cookiecutter.package_name }}/setup.cfg

```diff
@@ -1,62 +1,74 @@
 
 [options]
-# install_requires should be formatted as a semicolon-separated list, e.g.:
-# install_requires = astropy; scipy; matplotlib
-install_requires = {{ cookiecutter._install_requires }}
 zip_safe = False
-use_2to3 = False
+packages = find:
+python_requires = >={{ cookiecutter.minimum_python_version }}
+setup_requires = setuptools_scm
+install_requires = {{ cookiecutter._install_requires }}
 
+{% if cookiecutter.include_example_code == 'y' %}
 [options.entry_points]
 console_scripts =
-    {% if cookiecutter.include_example_code == 'y' %}
     astropy-package-template-example = packagename.example_mod:main
-    {% else %}
-    # astropy-package-template-example = packagename.example_mod:main
-    {% endif %}
+{% endif %}
+
+[options.extras_require]
+all =
+test =
+    pytest-astropy
+docs =
+    sphinx-astropy
 
 [options.package_data]
-* = *.c
 {{ cookiecutter.module_name }} = data/*
-{{ cookiecutter.module_name }}.tests = coveragerc
-
-[build_sphinx]
-source-dir = docs
-build-dir = docs/_build
-all_files = 1
-
-[build_docs]
-source-dir = docs
-build-dir = docs/_build
-all_files = 1
-
-[upload_docs]
-upload-dir = docs/_build/html
-show-response = 1
 
 [tool:pytest]
-minversion = 3.0
-norecursedirs = build docs/_build
+testpaths = "{{ cookiecutter.module_name }}" "docs"
+astropy_header = true
 doctest_plus = enabled
-addopts = -p no:warnings
-
-[ah_bootstrap]
-auto_use = True
+text_file_format = rst
+addopts = --doctest-rst
 
-[flake8]
-exclude = extern,sphinx,*parsetab.py,astropy_helpers,ah_bootstrap.py,conftest.py,docs/conf.py,setup.py
+[coverage:run]
+omit =
+    {{ cookiecutter.module_name }}/_{{ cookiecutter._parent_project }}_init*
+    {{ cookiecutter.module_name }}/conftest.py
+    {{ cookiecutter.module_name }}/*setup_package*
+    {{ cookiecutter.module_name }}/tests/*
+    {{ cookiecutter.module_name }}/*/tests/*
+    {{ cookiecutter.module_name }}/extern/*
+    {{ cookiecutter.module_name }}/version*
+    */{{ cookiecutter.module_name }}/_{{ cookiecutter._parent_project }}_init*
+    */{{ cookiecutter.module_name }}/conftest.py
+    */{{ cookiecutter.module_name }}/*setup_package*
+    */{{ cookiecutter.module_name }}/tests/*
+    */{{ cookiecutter.module_name }}/*/tests/*
+    */{{ cookiecutter.module_name }}/extern/*
+    */{{ cookiecutter.module_name }}/version*
 
-[pycodestyle]
-exclude = extern,sphinx,*parsetab.py,astropy_helpers,ah_bootstrap.py,conftest.py,docs/conf.py,setup.py
+[coverage:report]
+exclude_lines =
+    # Have to re-enable the standard pragma
+    pragma: no cover
+    # Don't complain about packages we have installed
+    except ImportError
+    # Don't complain if tests don't hit assertions
+    raise AssertionError
+    raise NotImplementedError
+    # Don't complain about script hooks
+    def main\(.*\):
+    # Ignore branches that don't pertain to this version of Python
+    pragma: py{ignore_python_version}
+    # Don't complain about IPython completion helper
+    def _ipython_key_completions_
```

Comment:
Why is IPython helper a problem for coverage test? Do we really need this?

Reply:
Packages can include this method on classes to help with IPython tab completion which isn't something that can be tested with pytest easily - I was just copying from the core package.